### PR TITLE
test: improve sample_degseq() tests

### PR DESCRIPTION
**Pull request by maelle**
- ensure we have tests for the examples in #1297
- put the tests in a test file named like the script.
- remove part of the test that I did not see any need for, see below

```r
  gc <- function(graph) {
    clu <- components(graph)
    induced_subgraph(graph, which(clu$membership == which.max(clu$csize)))
  }

  g <- gc(sample_gnp(1000, 2 / 1000))

  nG <- sample_degseq(degree(g), method = "simple")
  expect_that(degree(nG), equals(degree(g)))

  nG <- sample_degseq(degree(g), method = "vl")
  expect_that(degree(nG), equals(degree(g)))
  expect_true(is_connected(nG))
  expect_true(is_simple(nG))

  #####

  g <- sample_gnp(1000, 1 / 1000)

  nG <- sample_degseq(degree(g), method = "simple")
  expect_that(degree(nG), equals(degree(g)))

  g2 <- sample_gnp(1000, 2 / 1000, directed = TRUE)

  nG2 <- sample_degseq(degree(g, mode = "out"),
    degree(g, mode = "in"),
    method = "simple"
  )
  expect_that(degree(nG, mode = "out"), equals(degree(g, mode = "out")))
  expect_that(degree(nG, mode = "in"), equals(degree(g, mode = "in")))
```

### CODE REVIEW

**Comment by maelle** [comment]
will rebase once #1297 is merged.

**Comment by maelle** [comment]
@szhorvat do you feel strongly about any of the test cases I removed? See my first comment.

**Comment by szhorvat** [comment]
Is the code you quote in the top post what you removed? It's a bit hard to follow what's missing since code was moved between files.

It seems to me that the tests that you quote were all relevant. Why were they removed? I'm a bit lost.

**Comment by maelle** [comment]
> Why were they removed? I'm a bit lost.

Because I couldn't see what they brought based on the tests that are now in there, and they're a bit complicated (for instance, why is there a need to define the function called `gc()`).

**Comment by szhorvat** [comment]
These tests verify that the properties that the function guarantees actually hold: 

 - The degrees of the output graph are the same as the input
 - The graph is simple with methods that promise this
 - The graph is connected with methods that promise this

I don't mind a refactoring of the test, but all these properties _should_ be tested, ideally for all methods, including both the undirected and directed case.

**Comment by szhorvat** [comment]
P.S. It'd be good to rename the methods in this function quite soon, see #876. The current names are rather bad and confusing, I can't even remember which properties hold for which method without looking them up ... This is why I didn't list them.

**Comment by maelle** [comment]
ok, I'll add the tests back. thank you!

**Comment by maelle** [comment]
@szhorvat in

```
gc <- function(graph) {
    clu <- components(graph)
    induced_subgraph(graph, which(clu$membership == which.max(clu$csize)))
  }
```

what could be a more explicit name for gc? it'd help me (and future contributors I guess) understand the test.

**Comment by maelle** [comment]
this is nearly ready but I'd like to rename the helper function with your feedback. All tests are now in there, and they don't re-use the name "g" and "nG".

File: tests/testthat/test-games.R [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    degree(new_directed_graph, mode = "out"),
+    degree(directed_erdos_renyi, mode = "out")
+  )
+  expect_equal(
+    degree(new_directed_graph, mode = "in"),
+    degree(directed_erdos_renyi, mode = "in")
+  )
+})
+
+test_that("sample_degseq() works -- simple generator, connected", {
```
Comment (maelle) [inline tests/testthat/test-games.R:32]: the name of this test should also be updated once `gc()` has a better name.

**Comment by szhorvat** [comment]
> what could be a more explicit name for gc?

I'm sorry, I'm not paying attention ... "gc" refers to the giant component and actually we have a function for that now! It's called `largest_component()`.  The reason why it is used here is to create a degree sequence that is graphical and has a connected realization. This is done by creating a connected random graph, and taking its degrees. 

Let's do `g <- largest_component(sample_gnp(1000, 2 / 1000))`

**Comment by maelle** [comment]
mmmh why was this merged before the checks were run?!

**Comment by maelle** [comment]
@krlmlr did the Aviator settings somehow change?

**Comment by krlmlr** [comment]
Weird, let's keep an eye on that.
